Core/Business/Addon/Theme/Theme.php

```diff
@@ -75,4 +87,66 @@ public function onReset()
     {
         return true;
     }
+
+    public function setPropertiesFromJson()
+    {
+        $properties = $this->getJsonConfig('theme.json');
+
+        if (is_null($properties)) {
+            throw new \Exception("Missing theme.json for theme $this->directory", 1);
+        }
+
+        $properties->settings = $this->getJsonConfig('settings.json');
+
+        foreach ($properties as $prop => $value) {
+            $this->{$prop} = $value;
+        }
+
+        return $this;
+    }
+
+    public function initSettings()
+    {
+        $this->settings = new \stdClass();
+        $this->setAvailableLayouts();
+
+        return $this->saveSettings();
+    }
+
+    public function setAvailableLayouts()
+    {
+        $layouts = glob($this->directory.'templates/layouts/*.tpl');
+
+        $this->settings->available_layouts = [];
+        foreach ($layouts as $layout) {
+            $this->settings->available_layouts[] = basename($layout, '.tpl');
+        }
+
+        return $this;
+    }
+
+    public function setPageLayouts(array $layouts)
+    {
+        $this->settings->page_layouts = $layouts;
+        $this->saveSettings();
+    }
+
+    private function getJsonConfig($filename)
+    {
+        $file = $this->directory.'/config/'.$filename;
+
+        if (file_exists($file)) {
+            return json_decode(file_get_contents($file));
+        }
+
+        return null;
+    }
+
+    public function saveSettings()
```

Comment:
How do you intend to support having different settings for different shops using the same theme in multishop? I may have 3 columns on the homepage of shop 1 and only one column on the homepage of shop 2. Same with the other available settings.